### PR TITLE
gh-440: remove scipy as a test dependency

### PR DESCRIPTION
**Pull request by Saransh-cpp**
Remove scipy as a test dependency and replace the tests with scipy's internal tests.

Closes: #440

### CODE REVIEW

File: tests/core/test_array.py [hunk 1]
```diff
@@ -156,50 +151,47 @@ def test_trapezoid_product() -> None:
     np.testing.assert_allclose(s, 1.0)
 
 
-@pytest.mark.skipif(not HAVE_SCIPY, reason="test requires SciPy")
 def test_cumulative_trapezoid() -> None:
-    import scipy.integrate as spi
-
     # 1D f and x
```
Comment (Saransh-cpp) [inline tests/core/test_array.py:155]: scipy just compares the result with the actual values - https://github.com/scipy/scipy/blob/1d57f315d9454446d333d37d19f2a44f4baf35c9/scipy/integrate/tests/test_quadrature.py#L183

File: tests/core/test_algorithm.py [hunk 2]
```diff
@@ -1,33 +1,33 @@
...
-@pytest.mark.skipif(not HAVE_SCIPY, reason="test requires SciPy")
 def test_nnls(rng: np.random.Generator) -> None:
-    from scipy.optimize import nnls as nnls_scipy
-
-    # cross-check output with scipy's nnls
+    # check output
 
-    a = rng.standard_normal((100, 20))
-    b = rng.standard_normal((100,))
+    a = np.arange(25.0).reshape(-1, 5)
```
Comment (Saransh-cpp) [inline tests/core/test_algorithm.py:10]: inspired by - https://github.com/scipy/scipy/blob/1d57f315d9454446d333d37d19f2a44f4baf35c9/scipy/optimize/tests/test_nnls.py#L7

**Review by paddyroddy — APPROVED** [review]
Another one bites the dust 🩸

**Review by ntessore — APPROVED** [review]
Glad to see this go :)

**Pull request merged**